#149 implement update dockerfile to create virtual environment (#163)
TechnologyBrewery/habushu: README.md

```

### Leveraging the containerize-dependencies Goal to Prepare a Containerized Virtual Environment ###
If the execution is specified (does not run by default), the `containerize-dependencies` goal will collect a single `habushu` dependency specified the project's `pom.xml`, including all transitive habushu-packaged dependencies. After collecting the set of necessary dependencies, Habushu will copy the project files to the build directory under `containerize-support`, while preserving the original structure of the dependency modules to ensure that any path-based dependencies can be leveraged as-is. This directory can then be copied onto a Docker container and used to create a virtual environment capable of running the target Habushu project.

```xml
	<plugin>
		<groupId>org.technologybrewery.habushu</groupId>
		<artifactId>habushu-maven-plugin</artifactId>
		<configuration>

>           <dockerfile>${project.basedir}/src/main/docker/Dockerfile</dockerfile>

Reviewer: Think this needs to be spaced in more, looks like the indentation is off

Reviewer 2: It's because the surrounding lines use tabs, and this line uses spaces. Might just convert them all to spaces.

Author: yeah.. reformat the doc with spaces

	</plugin>
```

> To update the Dockerfile with containerize dependencies logic, you need to specify the Dockerfile location by adding the `dockerfile` tag under `configuration` and, also, to insert both the `#HABUSHU_BUILDER_STAGE` and `#HABUSHU_FINAL_STAGE` tags in the Dockerfile in the preferred location. 

Reviewer 3: ```suggestion
To update the `Dockerfile` with logic to leverage the monorepo dependencies, specify the `Dockerfile` location by adding the `dockerfile` tag under `configuration` and to insert both the `#HABUSHU_BUILDER_STAGE` and `#HABUSHU_FINAL_STAGE` tags in the `Dockerfile` in the preferred location. 
```

Reviewer 2: I'd prefer "logic to containerize the python application"

> Note: Miss adding the tags in the Dockerfile or configuring the dockerfile location will result the Dockerfile not being updated.

Reviewer 3: ```suggestion
Note: Not adding the tags in the `Dockerfile` or configuring the `dockerfile` location configuration will result in the Dockerfile not being updated.
```

> `#HABUSHU_BUILDER_STAGE` tag will be replaced by the habushu builder stage logic to, first, copy the generated dependencies files onto a Docker container and then to create the virtual environment for building the single habushu module.
> 
> `#HABUSHU_FINAL_STAGE` tag will be replaced by the habushu final stage logic to copy over the built wheel file to the final stage.

Reviewer 3: ```suggestion
The `#HABUSHU_BUILDER_STAGE` tag will be replaced with the habushu builder stage logic to, first, copy the generated dependencies files onto a Docker container and then to create the virtual environment for building the specified habushu module.

The `#HABUSHU_FINAL_STAGE` tag will be replaced with the habushu final stage logic to copy over the built virtual environment to the final stage.
```

```dockerfile
#HABUSHU_BUILDER_STAGE

FROM redhat/ubi9-minimal:latest

> RUN microdnf install -y python3.11

Reviewer 3: S: I thought that this was not needed, since the image declaration and setup will already be in the generated logic?

Reviewer 2: That would only be true for the builder stage.

Author: yeah, for the final stage, we only insert the copy over logic.

#HABUSHU_FINAL_STAGE

ENTRYPOINT ["/venv/bin/python3.11", "-m", "pets.main", "--enable_docs_url", "True"]
```

> 

Reviewer 3: C: we need to add the new configuration of `dockerfile` path to this README, preferably at the bottom of the configurations section

Author: the `dockerfile` path configuration is added to the readme but I can move it down to be more obvious.

Reviewer 3: We have a separate section that lists out all of the available configurations. We need to add a new section there for the dockerfile config
